src/vs/platform/storage/common/storage.ts

```diff
@@ -105,8 +105,10 @@ export interface IStorageService {
 	 *
 	 * @param target allows to define the target of the storage operation
 	 * to either the current machine or user.
+	 *
+	 * NOTE@coder: Add a promise so extensions can await storage writes.
 	 */
-	store(key: string, value: string | boolean | number | undefined | null, scope: StorageScope, target: StorageTarget): void;
+	store(key: string, value: string | boolean | number | undefined | null, scope: StorageScope, target: StorageTarget): Promise<void> | void;
```

Comment:
❓ Can we change to only return `Promise<void>`?

```suggestion
	store(key: string, value: string | boolean | number | undefined | null, scope: StorageScope, target: StorageTarget): Promise<void>;
```